src/test/ScrollTargetValidatorTest.cpp

```diff
@@ -0,0 +1,213 @@
+/************************************************************************/
+/*  Xania (M)ulti(U)ser(D)ungeon server source code                     */
+/*  (C) 2021 Xania Development Team                                     */
+/*  See the header to file: merc.h for original code copyrights         */
+/************************************************************************/
+#include "ScrollTargetValidator.hpp"
+#include "ObjectIndex.hpp"
+#include "ObjectType.hpp"
+#include "Target.hpp"
+
+#include <catch2/catch.hpp>
+#include <range/v3/algorithm/fill_n.hpp>
+
+namespace {
+
+const auto EmptyIndex = -1;
+const auto FireballIndex = 0;
+const auto EnchantIndex = 1;
+const auto ForgeIndex = 2;
+const auto FaerieFogIndex = 3;
+const auto DetectEvilIndex = 4;
+
+struct skill_type test_skill_table[5] = {
```

Comment:
unnecessary `struct` and maybe use `std::array` ? (though that makes constructing the validator need some gymnastics, so maybe not?)
```suggestion
const std::array<skill_type, 5> test_skill_table {
```